### PR TITLE
Show the impact changes at the end of the export_builder

### PR DESCRIPTION
**Pull request by ccomb**
[User story](https://www.notion.so/fd0068a7db8348c584cd53a3eea399f2?v=37bf8aeb782747c3bf744ecaeafefd04&p=1a75afd89ff94ac2817fe5bdebb37860&pm=s)

The check is done in the export_builder.py because it's easier here to compare the old and new impacts.
I arbitrarily chose N=5% of impact variation to stop the export and display the old new and percentage:
```
Impact htn-c of process domestic-gas-heat has evolved by more than 5.0%:
from 1.4114904768e-10 to 1.4961799054080002e-10.

Not recording builder.json
```

### CODE REVIEW

**Review by n1k0 — COMMENTED** [review]
Looks neat, how about integrating it to CI? 🤔

**Comment by ccomb** [comment]
Au lieu de devoir trancher sur une valeur incertaine, 5%,1%,  je me contente d'afficher tous les impacts modifiés, avec leur pourcentage d'évolution, uniquement pour ceux qui ont changé de plus de 0,1%.

Et ça exporte quand même le `builder.json`.

Ça affiche:
```
Impact etf of process domestic-gas-heat has evolved by 17%: from 0.4538383976439275 to 0.5321569155833066.
Impact etf-c of process domestic-gas-heat has evolved by 27%: from 0.4067007230421134 to 0.5169289547491491.
Impact htc-c of process domestic-gas-heat has evolved by 6.1%: from 3.099909664818703e-11 to 2.9119262618261356e-11.
Impact htn-c of process domestic-gas-heat has evolved by 0.19%: from 1.9582388280271603e-10 to 1.9544564537473592e-10.
######################################
Please review the impact changes above
######################################
```

**Comment by n1k0** [comment]
@ccomb just so you know, `builder.json` in this branch is now conflicting with the one from master.

File: data/food/export_agb/export_builder.py [hunk 1]
```diff
@@ -224,6 +208,27 @@ def search(name):
...
+    review = False
+    for p in builder:
+        for impact in builder[p]["impacts"]:
+            if old.get(p, {}).get(impact, {}):
+                percent_change = (
+                    100
+                    * abs(builder[p]["impacts"][impact] - old[p][impact])
+                    / old[p][impact]
+                )
+                if percent_change > 0.1:
```
Comment (n1k0) [inline data/food/export_agb/export_builder.py:222]: How about making this `0.1` value a command line parameter? 🤔

**Comment by ccomb** [comment]
Here is the output caused by the `patch wtu` branch 
```
Impact wtu of process egg has evolved by 72%: from 3.4034907 to 0.9685054495.
Impact wtu of process egg-organic has evolved by 76%: from 8.77810181 to 2.082957181.
Impact wtu of process egg-bleublanccoeur has evolved by 62%: from 1.958480459 to 0.7394206144.
Impact wtu of process milk has evolved by 32%: from 0.355589075 to 0.2417575942.
Impact wtu of process carrot has evolved by 82%: from 1.569591919 to 0.2798173697.
Impact wtu of process carrot-organic has evolved by 79%: from 0.2715674639 to 0.05832265459.
Impact wtu of process wheat has evolved by 36%: from 0.2186447912 to 0.1396706216.
Impact wtu of process wheat-organic has evolved by 82%: from 0.5390304088 to 0.0953745863.
Impact wtu of process flour has evolved by 35%: from 0.2590425677 to 0.1671542788.
Impact wtu of process flour-organic has evolved by 82%: from 0.6306898841159999 to 0.11577087785200002.
Impact wtu of process sunflower has evolved by 0.26%: from 0.1881891384 to 0.1886840977.
Impact wtu of process sunflower-organic has evolved by 11%: from 0.008456719901 to 0.009395157331.
Impact wtu of process rapeseed-organic has evolved by 1.4%: from 0.04637050692 to 0.04703642847.
Impact wtu of process tomato has evolved by 81%: from 1.335938865 to 0.2584271385.
Impact wtu of process tomato-organic has evolved by 80%: from 1.299676697 to 0.2551523711.
Impact wtu of process potato has evolved by 81%: from 0.5118653854 to 0.09972680971.
Impact wtu of process ground-beef has evolved by 30%: from 6.325161771 to 4.442890205.
Impact wtu of process ground-beef-organic has evolved by 9.1%: from 4.9856298363 to 4.5323238282.
Impact wtu of process beef-cattle has evolved by 31%: from 2.62735254 to 1.808934422.
Impact wtu of process beef-cattle-feedlot has evolved by 1.9%: from 5.118561913 to 5.02009286.
Impact wtu of process cull-cow-organic has evolved by 9.6%: from 2.044947351 to 1.847818606.
Impact wtu of process ground-beef-feedlot has evolved by 30%: from 6.325161771 to 4.442890205.
Impact wtu of process cooked-ham has evolved by 41%: from 4.972292304 to 2.943800913.
Impact wtu of process cooked-ham-organic has evolved by 50%: from 4.352764319559999 to 2.1926554799040003.
Impact wtu of process pig has evolved by 50%: from 2.003382851 to 0.9972243901.
Impact wtu of process pig-organic has evolved by 63%: from 1.696685829 to 0.6253702153.
Impact wtu of process lamb has evolved by 50%: from 9.244669879 to 4.667655903.
Impact wtu of process chicken-breast has evolved by 66%: from 6.37644955 to 2.142444098.
Impact wtu of process chicken-breast-organic has evolved by 75%: from 18.778149854 to 4.7515222424.
Impact wtu of process broiler has evolved by 70%: from 3.107326156 to 0.9171772996.
Impact wtu of process broiler-organic has evolved by 77%: from 15.50902646 to 3.526255444.
Impact wtu of process grape has evolved by 0.89%: from 0.02573007079 to 0.02550077126.
Impact wtu of process grape-organic has evolved by 2.7%: from 0.006261019973 to 0.006431073105.
Impact wtu of process mango has evolved by 83%: from 11.87278591 to 1.979929962.
Impact wtu of process sunfloweroil has evolved by 0.7%: from 0.6943598722 to 0.6992028204.
Impact wtu of process sunfloweroil-organic has evolved by 2.3%: from 0.25401544687744987 to 0.25994491649595.
Impact wtu of process zucchini has evolved by 82%: from 2.356732048 to 0.4354996348.
Impact wtu of process zucchini-organic has evolved by 82%: from 3.377227242 to 0.5971495246.
Impact wtu of process pear has evolved by 84%: from 6.276403083 to 1.032789749.
Impact wtu of process pear-organic has evolved by 84%: from 9.811280528 to 1.605095694.
Impact wtu of process peach has evolved by 83%: from 12.56300767 to 2.083835236.
Impact wtu of process peach-organic has evolved by 84%: from 21.30723512 to 3.478825701.
Impact wtu of process melon has evolved by 82%: from 4.234559782 to 0.7540589421.
Impact wtu of process melon-organic has evolved by 83%: from 4.938024656 to 0.8397746665.
Impact wtu of process butter has evolved by 31%: from 2.38022198 to 1.631194367.
Impact wtu of process butter-organic has evolved by 31%: from 2.38022198 to 1.631194367.
Impact wtu of process onion has evolved by 83%: from 2.419976851 to 0.4166246418.
Impact wtu of process eggplant has evolved by 82%: from 2.356732048 to 0.4354996348.
Impact wtu of process lettuce has evolved by 81%: from 1.717991514 to 0.3204930201.
Impact wtu of process frenchbean has evolved by 83%: from 5.228389232 to 0.8804390646.
Impact wtu of process durumwheat has evolved by 76%: from 1.830271926 to 0.4449605779.
Impact wtu of process steel has evolved by 0.16%: from 0.3597087043 to 0.3602792161.
Impact wtu of process glass has evolved by 0.72%: from 0.1988948574 to 0.2003323927.
Impact wtu of process cardboard has evolved by 0.16%: from 0.3235736496 to 0.3240757637.
Impact wtu of process paper has evolved by 0.19%: from 0.3689989281 to 0.3696900112.
Impact wtu of process pet has evolved by 0.17%: from 1.396396908 to 1.393982235.
Impact wtu of process aluminium has evolved by 7%: from 1.290304822 to 1.199935041.
Impact wtu of process lorry has evolved by 7.4%: from 0.0103829118 to 0.0111472208.
Impact wtu of process boat has evolved by 19%: from 0.0001996641126 to 0.0002379262796.
Impact wtu of process plane has evolved by 56%: from 0.006450266145 to 0.01006099252.
Impact wtu of process boat-cooling has evolved by 13%: from 0.0005869122882 to 0.0006646915631.
Impact wtu of process lorry-cooling has evolved by 6.2%: from 0.01277144344 to 0.01356289366.
Impact wtu of process domestic-gas-heat has evolved by 0.19%: from 0.0005870414538 to 0.0005881465245.
######################################
Please review the impact changes above
######################################
```

File: data/food/export_agb/export_builder.py (hunk 1 above)
Comment (paulboosz) [reply data/food/export_agb/export_builder.py:224]: it does display all differences of more than 0.1% so it'll catch differences of more than 3% or 5%

**Comment by n1k0** [comment]
@ccomb very nice! Would it be possible to sort the lines by largest diff descending, and format the strings so the values are easily readable as tabular data? eg.

```
trg   name                      diff   from          to
================================================================
wtu   tomato                     81%   1.335938865   0.258427138
wtu   potato                     81%   0.511865385   0.099726809
wtu   tomato-organic             80%   1.299676697   0.255152371
wtu   pig-organic                63%   1.696685829   0.625370215
wtu   cooked-ham-organic         50%   4.352764319   2.192655479
wtu   pig                        50%   2.003382851   0.997224390
wtu   lamb                       50%   9.244669879   4.667655903
wtu   cooked-ham                 41%   4.972292304   2.943800913
wtu   beef-cattle                31%   2.627352548   1.808934422
wtu   ground-beef                30%   6.325161771   4.442890205
wtu   ground-beef-feedlot        30%   6.325161771   4.442890205
wtu   cull-cow-organic          9.6%   2.044947351   1.847818606
wtu   ground-beef-organic       9.1%   4.985629836   4.532323828
wtu   beef-cattle-feedlot       1.9%   5.118561913   5.020092860
```

Maybe with some colors, eg. red for when diffs are >5%, etc

This could be another card for sure, but at least sorting could be a first useful step.

File: data/food/export_agb/export_builder.py (hunk 1 above)
Comment (n1k0) [reply data/food/export_agb/export_builder.py:224]: Though that could create lots of entries (noise), right? Anyway, I'm just trying to be helpful here, your call.

**Comment by ccomb** [comment]
I've just seen your suggestions, I thought I had merged. Since it's still open I'll try to do what you suggest

**Comment by ccomb** [comment]
(oops wrong button)

**Comment by ccomb** [comment]
Here is the real example output
```
Exported 47 ingredients to ../../../public/data/food/ingredients.json
=======================================================================================
Please review the impact changes below
=======================================================================================
trg    name                     diff                from                    to         
=======================================================================================
bvi    flour-organic            22.380934428222982  1.153213691             1.411313691
htc-c  carrot-organic           99.99999999999999   1.577471183e-10         0          
etf-c  tomato                   99.99999999999999   1.4686556131000001      0          
htc-c  tomato                   99.99999999999999   3.2479501073999997e-10  0          
htc-c  pig                      99.99999999999999   2.6211177582e-09        0          
htc-c  sunfloweroil             99.99999999999999   6.1383222846e-09        0          
htn-c  melon-organic            99.99999999999999   6.9849842428999996e-09  0          
htc-c  butter                   99.99999999999999   5.339558476e-09         0          
htc-c  butter-organic           99.99999999999999   5.339558476e-09         0          
htn-c  pet                      99.99999999999999   2.3636885388e-08        0          
etf-c  cooking                  99.99999999999999   0.7054322578000001      0          
htc-c  boat                     99.99999999999999   7.009799635e-12         0          
etf-c  boat-cooling             99.99999999999999   0.09755545345654        0          
etf-c  carrot-organic           100.0               1.29211471562           0          
htn-c  carrot-organic           100.0               2.46769729382e-09       0          
etf-c  wheat                    100.0               17.0995444342           0          
htc-c  wheat                    100.0               9.891593678e-10         0          
=======================================================================================
trg    name                     diff                from                    to         
=======================================================================================
Please review the impact changes above
=======================================================================================
Exported 77 processes to ../../../public/data/food/processes/builder.json
```

**Pull request merged**